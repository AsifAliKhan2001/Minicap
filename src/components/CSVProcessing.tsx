import React, { useState, useEffect } from "react";
import { View, Text, FlatList, StyleSheet } from "react-native";
import { readString } from "react-native-csv";
import * as FileSystem from "expo-file-system";
import { Asset } from "expo-asset";
import { Building } from "../viewmodels/Building"; // Import Building interface

const CSVProcessing = () => {
  // ✅ Use correct type for useState
  const [buildings, setBuildings] = useState<Building[]>([]);

  useEffect(() => {
    const loadCSV = async () => {
      try {
        const asset = Asset.fromModule(
          require("../../assets/BUILDING_LIST.csv")
        );
        await asset.downloadAsync();

        if (!asset.localUri) throw new Error("CSV file path is undefined");

        console.log("CSV File URI:", asset.localUri); // Debug log

        const csvData = await FileSystem.readAsStringAsync(asset.localUri);
        console.log("CSV Data Loaded:", csvData.substring(0, 200)); // Debug log

        // ✅ Now parse the CSV after ensuring data is available
        readString(csvData, {
          header: true,
          dynamicTyping: true,
          complete: (result: { data: Building[] }) => {
            setBuildings(result.data.slice(0, 5)); // ✅ Show only the first 5 buildings
          },
        });
      } catch (error) {
        console.error("❌ Error reading CSV file:", error);
      }
    };

    loadCSV();
  }, []);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Campus Buildings</Text>
      <FlatList
        data={buildings}
        keyExtractor={(item) => item.Building}
        renderItem={({ item }) => (
          <View style={styles.card}>
            <Text style={styles.heading}>{item.BuildingName}</Text>
            <Text>Campus: {item.Campus}</Text>
            <Text>Address: {item.Address}</Text>
          </View>
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: { flex: 1, padding: 20, backgroundColor: "#ffffff" },
  title: { fontSize: 24, fontWeight: "bold", marginBottom: 10, color: "#000" },
  card: {
    backgroundColor: "white",
    padding: 15,
    borderRadius: 10,
    marginBottom: 10,
    shadowColor: "#000",
    shadowOpacity: 0.1,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
    elevation: 2,
  },
  heading: { fontSize: 18, fontWeight: "bold", color: "#000" },
});

export default CSVProcessing;
