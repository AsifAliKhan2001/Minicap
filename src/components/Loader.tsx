import { View } from "react-native";
import { Spinner } from "@gluestack-ui/themed";
import { Dimensions } from "react-native";

type LoaderProps = {
  isLoading: boolean;
};

const Loader: React.FC<LoaderProps> = ({ isLoading }) => {
  const screenHeight = Dimensions.get("screen").height;

  if (!isLoading) return null;

  return (
    <View
      testID="loader"
      style={{
        position: "absolute",
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        height: screenHeight,
        backgroundColor: "rgba(0,0,0,0.6)",
        zIndex: 10,
      }}
    >
      <Spinner size="large" color="$white" />
    </View>
  );
};

export default Loader;
