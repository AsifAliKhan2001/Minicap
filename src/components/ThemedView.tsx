import { Box, BoxProps } from "@gluestack-ui/themed";
import { useThemeColor } from "@/hooks/useThemeColor";

export type ThemedViewProps = BoxProps & {
  lightColor?: string;
  darkColor?: string;
};

export function ThemedView({
  style,
  lightColor,
  darkColor,
  ...otherProps
}: ThemedViewProps) {
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "background"
  );

  return <Box bg={backgroundColor} style={style} {...otherProps} />;
}
