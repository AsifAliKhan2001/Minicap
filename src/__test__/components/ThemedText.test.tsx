import * as React from "react";
import renderer from "react-test-renderer";
import { ThemedText } from "../../components/ThemedText";

describe("ThemedText Component", () => {
  it("renders correctly with default type", () => {
    const tree = renderer
      .create(<ThemedText>Default Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with title type", () => {
    const tree = renderer
      .create(<ThemedText type="title">Title Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with defaultSemiBold type", () => {
    const tree = renderer
      .create(<ThemedText type="defaultSemiBold">SemiBold Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with subtitle type", () => {
    const tree = renderer
      .create(<ThemedText type="subtitle">Subtitle Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with link type", () => {
    const tree = renderer
      .create(<ThemedText type="link">Link Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles correctly", () => {
    const tree = renderer
      .create(
        <ThemedText style={{ fontSize: 20, fontWeight: "bold" }}>
          Custom Styled Text
        </ThemedText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("uses custom light and dark colors", () => {
    const tree = renderer
      .create(
        <ThemedText lightColor="blue" darkColor="red">
          Custom Colors
        </ThemedText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
