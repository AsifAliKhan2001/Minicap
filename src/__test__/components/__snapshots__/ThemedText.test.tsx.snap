// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText Component applies custom styles correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "fontSize": 20,
        "fontWeight": "bold",
      },
    ]
  }
>
  Custom Styled Text
</Text>
`;

exports[`ThemedText Component renders correctly with default type 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Default Text
</Text>
`;

exports[`ThemedText Component renders correctly with defaultSemiBold type 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      undefined,
      undefined,
      {
        "fontSize": 16,
        "fontWeight": "600",
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  SemiBold Text
</Text>
`;

exports[`ThemedText Component renders correctly with link type 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "color": "#0a7ea4",
        "fontSize": 16,
        "lineHeight": 30,
      },
      undefined,
    ]
  }
>
  Link Text
</Text>
`;

exports[`ThemedText Component renders correctly with subtitle type 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      undefined,
      undefined,
      undefined,
      {
        "fontSize": 20,
        "fontWeight": "bold",
      },
      undefined,
      undefined,
    ]
  }
>
  Subtitle Text
</Text>
`;

exports[`ThemedText Component renders correctly with title type 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      undefined,
      {
        "fontSize": 32,
        "fontWeight": "bold",
        "lineHeight": 32,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Title Text
</Text>
`;

exports[`ThemedText Component uses custom light and dark colors 1`] = `
<Text
  style={
    [
      {
        "color": "blue",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Custom Colors
</Text>
`;
